src/bin/redis_store_tester.rs

```diff
@@ -174,7 +185,7 @@ fn main() -> Result<(), Box<dyn core::error::Error>> {
 
                                 let res = store_clone.get_and_decode(data.clone()).await?;
                                 if let Some(existing_data) = res {
-                                    data.version = existing_data.version + 1;
+                                    data.version = existing_data.version;
```

Comment:
Didn't realize we tried to hack optimistic locking here. This change should prevent race conditions if the version was already up to date. Have you verified that this does not break the update logic of your tester?